### PR TITLE
persistence: add documentation

### PR DESCRIPTION
**Pull request by venkat1109**
This patch adds basic documentation for persistence. This includes how to setup config for different persistence implementations, how to add support for new database etc.

### CODE REVIEW

File: docs/persistence.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+```
+
+# Configuration
+## Common to all persistence implementations
+There are two major sub-subsystems within cadence that need persistence - cadence-core and visibility. cadence-core is
+the workflow engine that uses persistence to store state tied to domains, workflows, workflow histories, task lists 
+etc. cadence-core powers almost all of the cadence APIs. cadence-core could be further broken down into multiple 
+subs-systems that have slightly different persistence workload characteristics. But for the purpose of simplicity, we 
+don't expose these sub-systems today but this may change in future. Visibility is the sub-system that powers workflow 
+search. This includes APIs such as ListOpenWorkflows, ListClosedWorkflows or ListWorkflowsWithTag etc. Today, it is
```
Comment (vancexu) [inline docs/persistence.md]: Probably remove ListWorkflowsWithTag because we don't have it on Cassandra or MySQL.

File: docs/persistence.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+# Configuration
+## Common to all persistence implementations
+There are two major sub-subsystems within cadence that need persistence - cadence-core and visibility. cadence-core is
+the workflow engine that uses persistence to store state tied to domains, workflows, workflow histories, task lists 
+etc. cadence-core powers almost all of the cadence APIs. cadence-core could be further broken down into multiple 
+subs-systems that have slightly different persistence workload characteristics. But for the purpose of simplicity, we 
+don't expose these sub-systems today but this may change in future. Visibility is the sub-system that powers workflow 
+search. This includes APIs such as ListOpenWorkflows, ListClosedWorkflows or ListWorkflowsWithTag etc. Today, it is 
+possible to run a cadence server with cadence-core backed by one database and cadence-visibility backed by another kind 
+of database. The top level persistence configuration looks like the following:
```
Comment (vancexu) [inline docs/persistence.md]: Might worth to mention cadence support visibility on ElasticSearch which support search attributes and large scale (10M+ workflows)

Comment (vancexu) [reply docs/persistence.md]: And I will add a doc for elasticsearch setup

Comment (venkat1109) [reply docs/persistence.md]: I updated the docs to indicate that the full feature set of visibility is supported only in elastic search and that, a limited feature set is supported in cassandra /mysql. Would be awesome if you add the setup instructions / additional details for ES later.

File: docs/persistence.md (hunk 1 above)
Comment (venkat1109) [reply docs/persistence.md]: 👍

**Comment by dmetzgar** [comment]
Fixes #1841

**Pull request merged**